Added Services for pets

andrewdobra opened the pull request:
Added Services in a basic form, addition, view and a separate menu in the search section.

marcus-deszi committed:
Pet-profile, edit pet, delete pet. Problems with routing and with the checkboxes from add-edit-pet form; need help..

marcus-deszi committed:
Pet view, update, delete. Recommit

marcus-deszi committed:
Update lib/screens/pets/add-edit-pet/add-edit_pet_form.dart

Co-Authored-By: Denis Crăciunescu <dennis.craciunescu@gmail.com>

marcus-deszi committed:
Update lib/screens/pets/add-edit-pet/add-edit_pet_form.dart

Co-Authored-By: Denis Crăciunescu <dennis.craciunescu@gmail.com>

marcus-deszi committed:
Update lib/screens/pets/add-edit-pet/add-edit_pet_form.dart

Co-Authored-By: Denis Crăciunescu <dennis.craciunescu@gmail.com>

marcus-deszi committed:
Update lib/screens/pets/add-edit-pet/add-edit_pet_page.dart

Co-Authored-By: Denis Crăciunescu <dennis.craciunescu@gmail.com>

marcus-deszi committed:
modifications on view-edit-delete-pet + merge

marcus-deszi committed:
full merge

andrewdobra committed:
Added Services for pets

Added Services in a basic form, addition, view and a separate menu in the search section.

andrewdobra commented on lib/screens/home/search/service/service_search_page.dart:
```diff
@@ -0,0 +1,64 @@
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: _petsService
+          .getPetSittablePetsForOwnerId(_authService.currentUserUid),
```
Forgot to delete this, the search works though.

draciunescu replied:
Doubt that. 😄

andrewdobra replied:
For real though, it works somehow even with that thing over there, I'll remove it as it's garbage, probably from the category "don't know why it works but it does".

andrewdobra replied:
Actually I got it, it works because I just need to show the current users' pets in the list so if all of them are "petsittable" it works.

draciunescu commented on lib/screens/home/profile/profile_page.dart:
```diff
@@ -65,17 +78,26 @@ class _ProfilePageState extends State<ProfilePage> {
                   ),
-                  RaisedButton(
-                    child: Text("Add pet"),
+                  (currentUser.isServiceProvider == true)? RaisedButton(
```
Can be changed to `if (condition) RaisedButton`.

draciunescu commented on lib/screens/home/search/service/service_search_results.dart:
```diff
@@ -0,0 +1,33 @@
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: _serviceService.getServicesForPetType(forPet.type),
+      // ignore: missing_return
```
Can be removed.

andrewdobra replied:
What can be removed exactly? This part works as expected.

draciunescu replied:
The comment

draciunescu commented on lib/screens/services/add-service/add_service_form.dart:
```diff
@@ -0,0 +1,134 @@
+import 'package:pet_app/services/storage/storage_service.dart';
+import 'package:pet_app/services/services/services_service.dart';
+import 'package:pet_app/widgets/input_field.dart';
+//import 'package:pet_app/widgets/labeled_checkbox.dart';
```
Unused import.

draciunescu commented on lib/screens/services/add-service/add_service_form.dart:
```diff
@@ -0,0 +1,134 @@
+        _serviceService
+            .updateService(serviceObject);
+      });
+    }
```
Since `addServiceHandler` is async, it is possible for `updateService` to be called before the service was added.
Please fix this.

draciunescu commented on lib/screens/pets/add-pet/add_pet_form.dart:
```diff
@@ -128,10 +130,16 @@ class _AddPetFormState extends State<AddPetForm> {
+            .updatePet(petObject);
+      });
+    }
   }
```
Since `addPetHandler` is async, it is possible for `updatePet` to be called before the service was added.
Please fix this.

draciunescu commented on lib/screens/home/profile/profile_page.dart:
```diff
@@ -65,17 +78,26 @@ class _ProfilePageState extends State<ProfilePage> {
+                    onPressed: () {
+                      Navigator.of(context).pushNamed(MyServicesPage.routeName);
+                    },
+                  ): SizedBox(),
```
Can you extract all the `Service` related widgets to a separate Widget and only show that widget if the currentUser is a service provider? *to avoid duplicated condition

draciunescu commented on lib/screens/home/profile/profile_page.dart:
```diff
@@ -65,17 +78,26 @@ class _ProfilePageState extends State<ProfilePage> {
+                  SizedBox(
+                    height: (currentUser.isServiceProvider == true)? 10: 0,
                   ),
+                  (currentUser.isServiceProvider == true)? RaisedButton(
```
Can be changed to `if (condition) RaisedButton`.

marcus-deszi committed:
Reduce Scaffold by using functions which return the widgets needed

marcus-deszi committed:
being consistent: pushNamed instead of push

draciunescu committed:
Merge pull request #34 from deniskrr/pet-view-edit-delete

Pet-profile, edit pet, delete pet

andrewdobra committed:
Fixed a few things according to what was requested

I'm still not sure about the right approach regarding image upload but right now it works for me.

andrewdobra committed:
Merge branch 'develop' into andrew-dobra-la-butoane

andrewdobra committed:
Basic Service Profile + fixes

Added a basic Service profile based on the PetProfile, needs work as well as adding the chat launch when pressing the chat button.
